### ISSUE TITLE
compile issue

### ISSUE THREAD

**Issue by zTrix** [cloudflare/boringtun#108]
run `cargo install boringtun` produce the following output:

```
   Compiling chrono v0.4.9
   Compiling boringtun v0.2.0
error[E0308]: mismatched types
  --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/boringtun-0.2.0/src/device/tun_linux.rs:77:29
   |
77 |                 ifru_flags: IFF_TUN | IFF_NO_PI | IFF_MULTI_QUEUE,
   |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected i16, found i32

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
error: failed to compile `boringtun v0.2.0`, intermediate artifacts can be found at `/tmp/cargo-installRDLTdM`

Caused by:
  Could not compile `boringtun`.


```

Any hint on what's going on?

**Comment by vkrasnov** [comment]
Which platform is this?

Can you try replacing this with: `(IFF_TUN | IFF_NO_PI | IFF_MULTI_QUEUE) as _,` ?

**Comment by zTrix** [comment]
```
# uname -a
Linux zz 5.1.17-x86_64-linode128 #1 SMP PREEMPT Wed Jul 10 17:11:49 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by vkrasnov** [comment]
And have you tried the suggested fix?

**Comment by zTrix** [comment]
Just cloned this repo and build, everything works now.  

But the error happens when running `cargo install boringtun`.

No idea what happened.

**Comment by vkrasnov** [comment]
It appears that your kernel uses a slightly different version of libc

**Comment by tuxflo** [comment]
I had the same issue on a QNAP NAS device (custom kernel 4.14.24-qnap) and the fix above solved it for me.

**Comment by naftulikay** [comment]
When I attempt to `cargo install boringtun` within an Ubuntu 18.04 Bionic Docker container, I am running into this. On a previous version of Rust in around April 2019, the exact same environment built successfully, so I'm not sure what the issue is.

This sounds like it would require a source code fix, as was described above. I'm on x86_64, using the aforementioned Ubuntu Bionic Docker image.

**Comment by ossie-git** [comment]
Can confirm I ran into the same issue and that the suggested fix worked.  Steps to reproduce:

```
docker pull rust
docker run -ti rust bash
cargo install boringtun
```

**Comment by compuwar** [comment]
Same issue on CentOS 7.7
Linux host 3.10.0-1062.9.1.el7.x86_64 #1 SMP Fri Dec 6 15:49:49 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
CentOS Linux release 7.7.1908 (Core)

Suggested fix worked

**Comment by vkrasnov** [comment]
Fixed in release 0.3.0
